bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDSong.java

```diff
@@ -36,17 +33,19 @@
     private final int songId;
     private final String title;
     private final int track;
+    private final Optional<Integer> duration;
 
     public MPDSong(MPDResponse response) {
         Map<String, String> values = MPDResponseParser.responseToMap(response);
         filename = values.getOrDefault("file", "");
         album = values.getOrDefault("Album", "");
         artist = values.getOrDefault("Artist", "");
         name = values.getOrDefault("Name", "");
-        song = parseInteger(values.getOrDefault("Pos", "0"), 0);
-        songId = parseInteger(values.getOrDefault("Id", "0"), 0);
+        song = MPDResponseParser.parseInteger(values.getOrDefault("Pos", "0"), 0);
+        songId = MPDResponseParser.parseInteger(values.getOrDefault("Id", "0"), 0);
         title = values.getOrDefault("Title", "");
-        track = parseInteger(values.getOrDefault("Track", "-1"), -1);
+        track = MPDResponseParser.parseInteger(values.getOrDefault("Track", "-1"), -1);
+        duration = MPDResponseParser.parseInteger(values.get("Time"));
```

Comment:
Will change it to duration.

Reply:
Looking at this again.. That link states that time was deprecated in status, not current song.  The protocol doesn't seem to list or describe any fields for currentsong. 

I guess the appropriate thing to do is to move duration out of currentsong into status, parse a duration if it exists and if not fallback to time.

Reply:
I think it is a good a idea to first check for the duration in the status and then fallback to time. I would probably not consider the time in the current song, since this makes things a bit more complicated, but it would be ok for me. 
I still believe the time is deprecated (at least I see it in the section "Other Metadata").